### PR TITLE
Restructure metadata utils

### LINKED ISSUE #500: ProgressReport can count one changed spectrum multiple times
Currently ProgressReport.add_to_report, adds a count for each field that has been changed, instead of each spectrum that has been changed. 
This is a bit counterintuitive. For a filter like derive_adduct_from_name, this will result in a count for changes and a count for additions, since the compound_name field is changed, and a value was added to adduct. 

Suggested change:
Remove the differentation between addition and changes, and just keep the column changes, which can only be counted once. The column removed spectra will of course also stay.

### PR DESCRIPTION
**Pull request by niekdejonge**
Fixes #500 and restructure_metadata_utils has been reformated

In addition the ProcessingReport was changed to only show Changed spectrum or removed spectrum.

### CODE REVIEW

File: matchms/SpectrumProcessor.py [hunk 1]
```diff
@@ -268,6 +268,7 @@ def __str__(self):
     + ["clean_adduct",
        "derive_inchi_from_smiles",
        "derive_smiles_from_inchi",
```
Comment (niekdejonge) [inline matchms/SpectrumProcessor.py]: These are the only functional changes, the rest is restructuring of the files.

Comment (niekdejonge) [reply matchms/SpectrumProcessor.py]: @florian-huber I noticed I did not request a review yet, but the other PR's all build on this one. 
The main changes are moving functions to other files, to only functional change is here, changing the ProcessingReport output.

**Review by florian-huber — APPROVED** [review]
Thanks a lot @niekdejonge. Looks good to go for me!

**Comment by florian-huber** [comment]
@niekdejonge through the first few PRs I updated the `restructure_metadata_utils` branch to the current master branch. Also the PRs introduced several things that seem out of sync. So this and the the following PRs need to be adjusted regarding merge conflicts.

**Comment by florian-huber** [comment]
@niekdejonge I added a few things to the CHANGELOG, but probably missed some parts. Just add what you think is missing.

**Comment by niekdejonge** [comment]
> @niekdejonge I added a few things to the CHANGELOG, but probably missed some parts. Just add what you think is missing.

@florian-huber Thanks! I have also tried to add the changes to the changelog in #510 (not the right place, but was easier than updating all PR's individually) I will check that the changelog is formatted correctly in the end.

**Pull request merged**